blog/content/post/2021-12-02-meta-nezha.md

```diff
@@ -0,0 +1,162 @@
+---
+title: Introduction of Yocto meta layer for Nezha D1
+abstract: 'Presentation of current progress status with the support of the
+Nezha board in Yocto Project'
+cover: /img/nezha-logo.png
+author: cezary.sobczak
+layout: post
+published: true
+date: 2021-12-02
+archives: "2021"
+
+tags:
+  - risc-v
+  - sbc
+  - linux
+  - u-boot
+  - boot0
+  - opensbi
+  - bootloader
+  - logs
+  - SoC
+  - sdcard
+  - yocto
+  - layer
+  - meta
+categories:
+  - Firmware
+  - IoT
+  - Miscellaneous
+  - OS Dev
+  - App Dev
+  - Security
+  - Manufacturing
+
+---
+
+## Introduction
+
+As it was mentioned in first [post](https://blog.3mdeb.com/2021/2021-11-19-nezha-riscv-sbc-first-impression/)
+I'm interested in support for this board in Yocto Project, and this post will
+show you what I have achieved in this field.
+
+I didn't mention that earlier but the idea of creating the `meta-nezha` yocto
+layer was born in my head when I was thinking about the master thesis subject.
+So the whole project was created for educational purposes to share knowledge
+about Yocto and RISC-V.
+
+Whole code of the meta layer you can find at [github](https://github.com/Cezarus27/meta-nezha). It should be mentioned that `meta-nezha` is really early support so
+the recipe's code isn't that clear and for now, it isn't prepared to be
+upstreamed.
+
+## Nezha D1 meta layer construction
+
+Nezha D1 layer is using the following Yocto Project meta layers in `hardknott`
+version:
+
+* [poky](https://git.yoctoproject.org/git/poky)
+
+* [meta-openembedded](https://git.openembedded.org/meta-openembedded)
+
+* [meta-riscv](https://github.com/riscv/meta-riscv.git)
+
+The structure of a `meta-nezha` is divided into two main parts:
```

Comment:
@Cezarus27 maybe we could show this structure on some diagram? BTW we could explain why it is nice to seperate bsp vs distro(app) recipes in two layers.